syntax = "proto3";

package upload;

message UploadRequest {
    string folder = 1;
    string filename = 2;
    bytes content = 3;
    string content_type = 4;
}

message UploadResponse {
    string url = 1;
    string bucket = 2;
    string object_name = 3;
    string content_type = 4;
    int64 size = 5;
}

service UploadService {
    rpc UploadFile(UploadRequest) returns (UploadResponse);
}

